### PR TITLE
Add Foreman Architecture module

### PR DESCRIPTION
**Pull request by Manisha15**

### CODE REVIEW

**Comment by Manisha15** [comment]
@evgeni @mdellweg This is ready for review.

File: plugins/modules/foreman_architecture.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+DOCUMENTATION = '''
+---
+module: foreman_architecture
+short_description: Manage Foreman Architectures using Foreman API
+description:
+  - Create, Update and Delete Foreman Architectures using Foreman API
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+  - "apypie >= 0.0.1"
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]: Our wrapper requires 0.0.2 actually, but I think we've not been adding the version to the requirements in the modules themself.

File: plugins/modules/foreman_architecture.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+options:
+  name:
+    description: Name of architecture
+    required: true
+  operatingsystems:
+    description: List of operating systems the architecture should be assigned to
+    required: false
+    default: None
+  server_url:
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]: you can drop server_url/username/password/validate_verts and just add `extends_documentation_fragment: foreman` at the end of the documentation block. see https://github.com/theforeman/foreman-ansible-modules/pull/230/

File: plugins/modules/foreman_architecture.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            operatingsystems=dict(type='list'),
+            state=dict(default='present', choices=['present', 'absent']),
+        ),
+        supports_check_mode=True,
+    )
+    (entity_dict, state) = module.parse_params()
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]:
```suggestion
    entity_dict = module.clean_params()
```

see https://github.com/theforeman/foreman-ansible-modules/pull/304

File: plugins/modules/foreman_architecture.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    module.connect()
+
+    if 'operatingsystems' in entity_dict:
+        entity_dict['operatingsystems'] = module.find_resources('operatingsystems', entity_dict['operatingsystems'], thin=True)
+
+    entity = module.find_resource_by_name('architectures', name=entity_dict['name'], failsafe=True)
+
+    changed = module.ensure_resource_state('architectures', entity_dict, entity, state, name_map)
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]:
```suggestion
    changed = module.ensure_resource_state('architectures', entity_dict, entity, module.state, name_map)
```

see https://github.com/theforeman/foreman-ansible-modules/pull/304

File: plugins/modules/foreman_architecture.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    'operatingsystems': 'operatingsystem_ids',
+}
+
+
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            operatingsystems=dict(type='list'),
+            state=dict(default='present', choices=['present', 'absent']),
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]: as you don't have any "special" states, you can omit this line, as `ForemanEntityApypieAnsibleModule` already defines a state:

https://github.com/theforeman/foreman-ansible-modules/blob/703baad75c867c266a3dc71980f2e792925b3d41/plugins/module_utils/foreman_helper.py#L278-L285

File: plugins/modules/foreman_architecture.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    username: "admin"
+    password: "secret"
+    state: present
+
+- name: "Update an Architecture"
+  foreman_architecture:
+    name: "architecture_name"
+    operatingsystems:
+      - "operatingsystem_name3"
+      - "operatingsystem_name4"
```
Comment (sean797) [inline plugins/modules/foreman_architecture.py]: Please use meaningful data in all the examples e.g `name: x86_64`

File: plugins/modules/foreman_architecture.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            operatingsystems=dict(type='list'),
+            state=dict(default='present', choices=['present', 'absent']),
+        ),
+        supports_check_mode=True,
+    )
+    (entity_dict, state) = module.parse_params()
+
+    module.connect()
+
+    if 'operatingsystems' in entity_dict:
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]: `if not module.desired_absent` should be added here, see #304

**Pull request merged**